ten/one_david.py

```diff
@@ -0,0 +1,76 @@
+class BinaryHeap:
+    
+    def __init__(self, min=True):
+        self.arr = []
+        self.size = 0
+        self._sign = 2*min - 1
+        
+    def _get_val(self, ind):
+        return self.arr[ind][0]
+        
+    def push(self, val, data=None):
+        val *= self._sign
+        if self.size == len(self.arr):
+            self.arr.append((val, data))
+        else:
+            self.arr[self.size] = (val, data)
+        self.size += 1
+        
+        cur_ind = self.size - 1
+        while cur_ind > 0:
+            par_ind = cur_ind // 2
+            if self._get_val(par_ind) > self._get_val(cur_ind):
```

Comment:
this is where you would use `self.less_than(self._get_val(par_ind), self._get_val(cur_ind))` if you used that approach.